it/uk/gov/hmrc/soletraderidentificationfrontend/api/controllers/JourneyControllerISpec.scala

```diff
@@ -73,15 +93,35 @@ class JourneyControllerISpec extends ComponentSpecHelper with JourneyStub with S
       "signOutUrl" -> testSoleTraderJourneyConfig.pageConfig.signOutUrl,
       "accessibilityUrl" -> testSoleTraderJourneyConfig.pageConfig.accessibilityUrl
     )
-    "redirect to Capture Full Name Controller" in {
-      stubAuth(OK, successfulAuthResponse())
-      stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
 
-      lazy val result = post("/sole-trader-identification/api/sole-trader-journey", testSoleTraderJourneyConfigJson)
+    "returns json containing the url to Capture Full Name Controller" when {
+
+      "an optFullNamePageLabel field is not provided" in {
+        stubAuth(OK, successfulAuthResponse())
+        stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+
+        lazy val result = post("/sole-trader-identification/api/sole-trader-journey", testSoleTraderJourneyConfigJson)
+
+        (result.json \ "journeyStartUrl").as[String] must include(controllerRoutes.CaptureFullNameController.show(testJourneyId).url)
+
+        await(journeyConfigRepository.findById(testJourneyId)) mustBe Some(testSoleTraderJourneyConfig.copy(businessVerificationCheck = false))
+      }
+
+      "an optFullNamePageLabel field is provided" in {
+        stubAuth(OK, successfulAuthResponse())
+        stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+
+        post(
+          uri = "/sole-trader-identification/api/sole-trader-journey",
+          json = testSoleTraderJourneyConfigJson + ("optFullNamePageLabel" -> JsString(testFullNamePageLabel))
+        )
 
-      (result.json \ "journeyStartUrl").as[String] must include(controllerRoutes.CaptureFullNameController.show(testJourneyId).url)
+        val expectedSoleTraderJourneyConfig = testSoleTraderJourneyConfig
+          .copy(businessVerificationCheck = false)
+          .copy(pageConfig = testSoleTraderPageConfig.copy(optFullNamePageLabel = Some(testFullNamePageLabel)))
 
-      await(journeyConfigRepository.findById(testJourneyId)) mustBe Some(testSoleTraderJourneyConfig.copy(businessVerificationCheck = false))
+        await(journeyConfigRepository.findById(testJourneyId)) mustBe Some(expectedSoleTraderJourneyConfig)
+      }
```

Comment:
Is this not the same as the test you added above?

Reply:
Which one? They are 3 one for each journey 
/sole-trader-identification/api/journey
/sole-trader-identification/api/sole-trader-journey
/sole-trader-identification/api/individual-journey
only the url is different

Reply:
Ohhh okay, makes sense. My mistake